spec/services/location_data_service_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
+RSpec.describe LocationDataService, type: :service do
+  subject(:service) { described_class.new }
+
+  before do
+    stub_request(:get, "https://api-na.hosted.exlibrisgroup.com/almaws/v1/conf/libraries")
+      .with(
+        headers: {
+          'Accept' => 'application/json',
+          'Authorization' => 'apikey TESTME'
+        }
+      )
+      .to_return(
+        status: 200,
+        headers: { "content-Type" => "application/json" },
+        body: file_fixture("alma/libraries.json")
+      )
+
+    stub_request(:get, "https://api-na.hosted.exlibrisgroup.com/almaws/v1/conf/libraries/arch/locations")
+      .with(
+        headers: {
+          'Accept' => 'application/json',
+          'Authorization' => 'apikey TESTME'
+        }
+      )
+      .to_return(
+        status: 200,
+        headers: { "content-Type" => "application/json" },
+        body: file_fixture("alma/locations1.json")
+      )
+
+    stub_request(:get, "https://api-na.hosted.exlibrisgroup.com/almaws/v1/conf/libraries/main/locations")
+      .with(
+        headers: {
+          'Accept' => 'application/json',
+          'Authorization' => 'apikey TESTME'
+        }
+      )
+      .to_return(
+        status: 200,
+        headers: { "content-Type" => "application/json" },
+        body: file_fixture("alma/locations2.json")
+      )
+  end
+
+  describe "#delete_existing_and_repopulate" do
+    before do
+      # Setup dummy records to test if existing records are deleted
+      test_library = Locations::Library.create(code: 'test', label: 'test')
+      Locations::HoldingLocation.new(code: 'test', label: 'test') do |test_location|
+        test_location.library = test_library
+        test_location.save
+      end
+    end
+
+    it "deletes exsiting data and populates library and location data from Alma" do
+      LocationDataService.delete_existing_and_repopulate
+      library_record = Locations::Library.find_by(code: 'main')
+      location_record1 = Locations::HoldingLocation.find_by(code: 'main$stacks')
+      location_record2 = Locations::HoldingLocation.find_by(code: 'arch$reserves')
+
+      expect(Locations::Library.count).to eq 2
+      expect(Locations::HoldingLocation.count).to eq 13
+      expect(library_record.label).to eq 'Main Library'
+      expect(location_record1.label).to eq 'Main Library Stacks'
```

Comment:
I just looked over the notes from the meeting; we talked about doing "Main Library - Stacks" @mzelesky can you confirm?

Reply:
I can't figure out where this comes from, though. Does it just come from Alma directly?

Reply:
Okay. In the notes we said the location label should be `"#{library.name} - #{location.name}"` -- but looking at the fixture data it seems like the location names sometimes include the library name? But sometimes they dont, e.g. "Microform Room". So I think we do need to use both, and potentially change some of the location names in alma to not also include the library names.

Reply:
Thanks @mzelesky

Reply:
The fixtures in this PR were drawn from the Alma API examples and don't reflect PUL names. I didn't concatenate the library names because as Mark suggested, they need to be cleaned up. E.g. the firestone stacks location would be rendered as. "Firestone Library - Firestone Library",

Reply:
I think showing the data in the form it will actually display could motivator for those tasked with cleaning it, though.